src/main/java/org/commcare/formplayer/objects/SerializableMenuSession.java

```diff
@@ -1,30 +1,61 @@
-package org.commcare.formplayer.repo;
+package org.commcare.formplayer.objects;
 
 import org.commcare.formplayer.session.MenuSession;
+import org.hibernate.annotations.GenericGenerator;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+import javax.persistence.*;
 
 /**
  * Created by willpride on 8/1/16.
  */
+
+@Entity
+@Table(name="menu_sessions")
+@EntityListeners(AuditingEntityListener.class)
 public class SerializableMenuSession {
```

Comment:
based on how the update was written (https://github.com/dimagi/formplayer/blob/0a6850fdbeb61d8c5dedcda4c4c338e350fbd275/src/main/java/org/commcare/formplayer/repo/impl/PostgresMenuSessionRepo.java#L49-L51) it looks like the only thing that can change / needs to be saved to the DB is the `commcareSession`. 

If that's the case you can annotate all the other fields with `@Column(updatable=false)`. You can also remove all the "setters" for those fields.

Side note: you could delete all the getters and setters and then let [lombok](https://projectlombok.org/features/GetterSetter) generate them for you (annotate the class with `@Getter` and only annotate `commcareSession` with `@Setter`)

Reply:
> If that's the case you can annotate all the other fields with @Column(updatable=false). You can also remove all the "setters" for those fields.

Just curious, what't the advantage of marking something `updatable=false`?

Reply:
> Just curious, what't the advantage of marking something updatable=false?

it won't be included in the SQL update statement when saving is better for network and DB